server/v2/cometbft/abci.go

```diff
@@ -315,8 +315,12 @@ func (c *Consensus[T]) PrepareProposal(
 		return nil, errors.New("PrepareProposal called with invalid height")
 	}
 
-	decodedTxs := make([]T, len(req.Txs))
-	for i, tx := range req.Txs {
+	if c.prepareProposalHandler == nil {
+		return nil, errors.New("no prepare proposal function was set")
+	}
+
+	var decodedTxs []T
```

Comment:
we should preallocate here as we know the length

Reply:
Thought the same but we skip undecodable txs, so the index would then be nil

Reply:
The only case we can't decode is if it's a vote exetension or some other data type of the state machine right?

Reply:
Should be yeah, if the proposer puts gibberish in its prepare proposal.

Reply:
need to remove the decoding part and make it part of the handler. it wont work this way, and requiring users to modify the decoder feels weird

Reply:
makes sense to move it to handler.